src/thunderbots/software/ai/hl/stp/play/stop_play.cpp

```diff
@@ -22,40 +24,105 @@ bool StopPlay::invariantHolds(const World &world) const
+    float radius = sqrt(2) * world.field().friendlyDefenseArea().width();
 
     do
     {
-        move_tactic_1->updateParams(stop_positions.at(0), Angle::zero(), 0);
-        move_tactic_2->updateParams(stop_positions.at(1), Angle::zero(), 0);
-        move_tactic_3->updateParams(stop_positions.at(2), Angle::zero(), 0);
-        move_tactic_4->updateParams(stop_positions.at(3), Angle::zero(), 0);
-        move_tactic_5->updateParams(stop_positions.at(4), Angle::zero(), 0);
-        move_tactic_6->updateParams(stop_positions.at(5), Angle::zero(), 0);
-
-        // yield the Tactics this Play wants to run, in order of priority
-        yield({move_tactic_1, move_tactic_2, move_tactic_3, move_tactic_4, move_tactic_5,
-               move_tactic_6});
+        // goalie tactic
+        auto enemy_threats = Evaluation::getAllEnemyThreats(
+            world.field(), world.friendlyTeam(), world.enemyTeam(), world.ball(), false);
+
+        goalie_tactic->updateParams(
+            world.ball(), world.field(), world.friendlyTeam(), world.enemyTeam(),
+            enemy_threats.empty() ? std::nullopt
+                                  : std::make_optional(enemy_threats.at(0)));
+        std::vector<std::shared_ptr<Tactic>> result = {goalie_tactic};
+
+        // a unit vector from the center of the goal to the ball
+        Vector goal_to_ball_unit_vector =
+            (world.field().friendlyGoal() - world.ball().position()).norm();
+
+        // defense_point_center is a point on the semicircle around the friendly defense
+        // area, that can block the direct path from the ball to the net.
+        // robot_positioning_unit_vector will be used to position the robots on the
+        // tangent line on that semicircle
+        Vector robot_positiioning_unit_vector = goal_to_ball_unit_vector.perp();
+        Point defense_point_center =
+            world.field().friendlyGoal() - radius * goal_to_ball_unit_vector;
+
+        // position robots on either side of the "defense point"
+        Point defense_point_left = defense_point_center + robot_positiioning_unit_vector *
+                                                              2 * ROBOT_MAX_RADIUS_METERS;
+        Point defense_point_right =
+            defense_point_center -
+            robot_positiioning_unit_vector * 2 * ROBOT_MAX_RADIUS_METERS;
+
+        move_tactics.at(0)->updateParams(
+            defense_point_left,
+            (world.ball().position() - defense_point_left).orientation(), 0);
+        move_tactics.at(1)->updateParams(
+            defense_point_right,
+            (world.ball().position() - defense_point_right).orientation(), 0);
+
+        // move the robots in a triangle around the ball, facing the ball, this behaviour
+        // is very similar to the example play when there are only 3 robots, except spaced
+        // 0.5m away from the ball
+        Angle angle_between_robots = Angle::full() / 3;
```

Comment:
Can we keep all the robots closer together, between the ball and net?

Reply:
fixed